sklego/linear_model.py

```diff
@@ -121,18 +188,78 @@ def fit(self, X, y):
         return self
 
     def predict(self, X):
-        """
-        Predict using ProbWeightRegression.
+        """Predict target values for `X` using fitted estimator by multiplying `X` with the learned coefficients.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            The data to predict.
 
-        :param X: array-like, shape=(n_columns, n_samples, ) training data.
-        :return: Returns an array of predictions shape=(n_samples,)
+        Returns
+        -------
+        array-like of shape (n_samples,)
+            The predicted data.
         """
         X = check_array(X, estimator=self, dtype=FLOAT_DTYPES)
         check_is_fitted(self, ["coefs_"])
         return np.dot(X, self.coefs_)
 
 
 class DeadZoneRegressor(BaseEstimator, RegressorMixin):
+    r"""The `DeadZoneRegressor` estimator implements a regression model that incorporates a _dead zone effect_ for
+    improving the robustness of regression predictions.
+
+    The dead zone effect allows the model to reduce the impact of small errors in the training data on the regression
+    results, which can be particularly useful when dealing with noisy or unreliable data.
+
+    The estimator minimizes the following loss function using gradient descent:
+
+    $$\frac{1}{n} \sum_{i=1}^{n} \text{deadzone}\left(\left|X_i \cdot w - y_i\right|\right)$$
+
+    where:
+
+    $$\text{deadzone}(e) =
+    \begin{cases}
+    e & \text{if } e > \text{threshold} \text{ and effect="linear"} \\
+    e^2 & \text{if } e > \text{threshold} \text{ and effect="quadratic"} \\
+    0 & \text{otherwise}
+    \end{cases}
+    $$
+
+    Parameters
+    ----------
+    threshold : float, default=0.3
+        The threshold value for the dead zone effect.
+    relative : bool, default=False
+        If True, the threshold is relative to the target value. Namely the _dead zone effect_ is applied to the
+        relative error between the predicted and target values.
+    effect : Literal["linear", "quadratic"], default="linear"
```

Comment:
I didn't documented `effect="constant"` because it breaks during `.fit()` (as `deadzone(errors)` would return None). 
I am not sure what the expected behaviour is: I would imagine the loss to be 0 within threshold and some constant value outside of it, but such value is not provided anywhere